### DISCUSSION TITLE
pb_hooks database query not working

### DISCUSSION THREAD

**Discussion by H2OSANI** [pocketbase/pocketbase#4560] [Q&A] [answered]
Hi i am trying to write an import script for my database but making queries in the pb_hooks seems to be more difficult than i thought. The Querybuilder does not find a record even if one exists. I also tried _$dbx.exp_ in the where clause. The ticker field is a string field.
`$app.rootCmd.addCommand(new Command({
    use: "import_fundamentals",
    run: (cmd, args) => {

        let rawData = require(args[0]);

        rawData.forEach(element => {
            const stock = $app.dao().db()
                .select("id", "ticker")
                .from("stock")
                .where($dbx.like("ticker", element.symbol))
        });
    },
}));`

**Comment by ganigeorgiev** [comment, marked as answer]
The above query never executes. To run raw queries, please refer to https://pocketbase.io/docs/js-database/#executing-queries.

Here is a slightly modified version of your code:

```js
$app.rootCmd.addCommand(new Command({
    use: "import_fundamentals",
    run: (cmd, args) => {
        let rawData = require(args[0]);

        rawData.forEach(element => {
            const stock = new DynamicModel({
                // describe the shape of the data (used also as initial values)
                "id":     "",
                "ticker": "",
            })

            $app.dao().db()
                .select("id", "ticker")
                .from("stock")
                .where($dbx.like("ticker", element.symbol))
                .limit(1)
                .one(stock)

            console.log(stock)
        });
    },
}));
```

Note that if "stock" is a collection with records, instead of manually constructing the query you can also use the records retrieval helpers listed in https://pocketbase.io/docs/js-records/#fetch-records.
